### ISSUE TITLE
`non-type template parameter` should be `non-type \grammarterm{template-parameter}` more consistently

### ISSUE THREAD

**Issue by Tsche** [cplusplus/draft#7065]
Currently the C++ standard uses `non-type \grammarterm{template-parameter}` 15 times, but uses `non-type template parameter` in the following 11 places:

[[lex.ext]/5](https://standards.pydong.org/c++/lex.ext#5)
[[dcl.type.auto.deduct]/2.4](https://standards.pydong.org/c++/dcl.type#auto.deduct-2.4)
[[temp.param]/11](https://standards.pydong.org/c++/temp.param#11)
[[temp.dep.type]/2](https://standards.pydong.org/c++/temp.dep.type#2)
[[temp.dep.constexpr]/2.3](https://standards.pydong.org/c++/temp.dep.constexpr#2.3)
[[temp.deduct.general]/11.8](https://standards.pydong.org/c++/temp.deduct.general#11.8)
[[temp.deduct.call]/1](https://standards.pydong.org/c++/temp.deduct.call#1)
[[temp.deduct.type]/13](https://standards.pydong.org/c++/temp.deduct.type#13)
[[temp.over.link]/6.3](https://standards.pydong.org/c++/temp.over.link#6.3)
[[rand.eng.general]/6](https://standards.pydong.org/c++/rand.eng.general#6)
[[rand.adapt.general]/6](https://standards.pydong.org/c++/rand.adapt.general#6)

I believe it would be nice to use `\grammarterm{template-parameter}` in most if not all of those places.

**Comment by frederick-vs-ja** [comment]
It seems better to me to properly define _non-type template parameter_, _type template parameter_, and _template template parameter_ in [basic.scope.temp] and then use "plain English" forms.

**Comment by jensmaurer** [comment]
The cleanup is part of [P2841](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2024/p2841r3.pdf).

**Comment by eisenwave** [comment]
Now that P2841 has been merged, is there anything left for us to do?

We may be able to close this.

**Comment by frederick-vs-ja** [comment]
Yeah. The paper defined _meow_ template parameters and #7690 should be considered to have resolved this issue.
